### ISSUE TITLE
TypeError: Cannot read properties of null (reading 'useContext')

### ISSUE THREAD

**Issue by wdavidw** [dc7290/next-export-optimize-images#796]
**Describe the bug**

When running `npx next build`, the following error is thrown:

```
   Creating an optimized production build ...
 ✓ Compiled successfully
 ✓ Linting and checking validity of types    
 ✓ Collecting page data    
   Generating static pages (4/5)  [=   ] 
TypeError: Cannot read properties of null (reading 'useContext')
    at exports.useContext (/Users/gollum/project/node_modules/react/cjs/react.production.min.js:24:118)
    at /Users/gollum/project/node_modules/next/dist/client/image-component.js:243:47
    at nM (/Users/gollum/project/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:60916)
    at nN (/Users/gollum/project/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64674)
    at nI (/Users/gollum/project/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:46806)
    at nM (/Users/gollum/project/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:60923)
    at nN (/Users/gollum/project/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64674)
    at nB (/Users/gollum/project/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:67657)
    at nF (/Users/gollum/project/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:66824)
    at nN (/Users/gollum/project/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:12:64990)
```

**To Reproduce**

I have followd the getting started page and dropped a default configuration.

**Additional context**

This is an ESM project.

**Comment by wdavidw** [comment]
Addtionnal information, the error happens both for CommonJS and ESM project. However the next.config.mjs file is written as ESM. It is based on the TailwindUI Commit template and uses MDX.

**Comment by JoffreyB31** [comment]
Same here

**Comment by dc7290** [comment]
Maybe this is happening since Next.js v14.

**Comment by dc7290** [comment]
It seems to occur probably in 14.1.0 or later.

https://github.com/vercel/next.js/issues/49355

**Comment by hauerm** [comment]
@dc7290 FYI: it stopped working with next@13.5.0. next@13.4.19 works.

**Comment by dc7290** [comment]
Ref: https://github.com/dc7290/next-export-optimize-images/issues/820

**Comment by dc7290** [comment]
Fixed release in v4.0.0!
